zerver/migrations/0520_user_presence_backfill_last_update_id_to_0.py

```diff
@@ -0,0 +1,39 @@
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def backfill_user_presence_last_update_id(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    UserPresence = apps.get_model("zerver", "UserPresence")
+
+    max_id = UserPresence.objects.aggregate(models.Max("id"))["id__max"]
+    if max_id is None:
+        # Nothing to do if there are no rows yet.
+        return
+
+    BATCH_SIZE = 10000
+    lower_bound = 0
+
+    while lower_bound < max_id:
```

Comment:
As noted in previous migrations, we should run past `max_id` by a `BATCH_SIZE` to deal with the possibility of races.

Reply:
Should we have some kind of semgrep rule for this then? These migrations are typically written with the idea that they're supposed to run after the server already loaded the new backend code that ensures new objects get created correctly; so running past `Max` is not needed under that assumption, so it feels pretty unintuitive to it. We have some other recent migrations that don't run past the `Max`, so we're not successfully enforcing this margin of error without some lint rule it seems.

Reply:
Tweaked both migrations to do the `max_id += BATCH_SIZE / 2` trick. Had to also update the `PresenceSequence` migration to filter out realms that already have the object:
```
        realm_ids = Realm.objects.filter(
            id__gt=lower_bound,
            id__lte=lower_bound + BATCH_SIZE,
            # Filter to realm whose PresenceSequence does not exist, to avoid
            # running into IntegrityError by trying to create duplicate PresenceSequence.
            presencesequence=None,
        ).values_list("id", flat=True)
```

Reply:
In most cases, the failure mode of a race is just that the next migration (to enforce the backfill has completed) fails, so it's generally "safe" in that respect. But yeah, that seems like a good approach.